From 4102271bbd16d93984b54a3435995259e5641e1c Mon Sep 17 00:00:00 2001
From: Albert Astals Cid <albert.astals.cid@kdab.com>
Date: Fri, 7 May 2021 16:38:13 +0200
Subject: [PATCH 195/196] Make sure _q_printerChanged is called even if only
 pdf printer is available

Otherwise things like the pagesRadioButton don't get properly disabled

Change-Id: Idc0693b6747f13da5afb22e2187af310a5b454f8
Pick-to: 6.1 5.15
Reviewed-by: Andy Shaw <andy.shaw@qt.io>
(cherry picked from commit 973700c54678803a2600dfa60112f29f15705197)
---
 src/printsupport/dialogs/qprintdialog_unix.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/printsupport/dialogs/qprintdialog_unix.cpp b/src/printsupport/dialogs/qprintdialog_unix.cpp
index bf77666c41..c5b845f4ec 100644
--- a/src/printsupport/dialogs/qprintdialog_unix.cpp
+++ b/src/printsupport/dialogs/qprintdialog_unix.cpp
@@ -1184,6 +1184,8 @@ void QUnixPrintWidgetPrivate::updateWidget()
             widget.printers->insertSeparator(widget.printers->count());
         widget.printers->addItem(QPrintDialog::tr("Print to File (PDF)"));
         filePrintersAdded = true;
+        if (widget.printers->count() == 1)
+            _q_printerChanged(0);
     }
     if (!printToFile && filePrintersAdded) {
         widget.printers->removeItem(widget.printers->count()-1);
-- 
2.31.1

